drivers/gles3/storage/material_storage.cpp

```diff
@@ -1108,10 +1109,10 @@ MaterialStorage::MaterialStorage() {
 
 	static_assert(sizeof(GlobalShaderUniforms::Value) == 16);
 
-	global_shader_uniforms.buffer_size = MAX(4096, (int)GLOBAL_GET("rendering/limits/global_shader_variables/buffer_size"));
```

Comment:
The property hint range in the project settings should be changed accordingly to match:

https://github.com/godotengine/godot/blob/6b38be699d56eec119d955c700c853ba3053cca9/servers/rendering_server.cpp#L3613

I'd suggest `16,1048576,1`.

Reply:
The minimum is still 4096 for the RD backend.

Reply:
I see, but if neither backend can use values between 1 and 16, I'd make the minimum value in the inspector 16.

Reply:
This PR should actually edit `godot/servers/rendering_server.cpp` to make the default value to `4096` instead of the (new) wrong value `65536`.

Reply:
khronos suggests 16K https://www.khronos.org/opengl/wiki/GLAPI/glGet
```
GL_MAX_UNIFORM_BLOCK_SIZE (integer, minimum is 16384)
The maximum size in basic machine units of a uniform block.
```

Reply:
> This PR should actually edit godot/servers/rendering_server.cpp to make the default value to 4096 instead of the (new) wrong value 65536.

I'm not sure what you mean. 4096 is the minimum value when using the RD backend, 65536 is the default. I don't think the default should be lowered to the minimum value.

> khronos suggests 16K https://www.khronos.org/opengl/wiki/GLAPI/glGet

16384 is the specification's required minimum size for the max size of a uniform block in bytes. I.e. all implementations of OpenGL 3.3 have to support a block size of at least 16384 bytes. But they can support larger block sizes if they want. 

This project setting value is the maximum number of uniforms in the uniform array. So they are not equivalent

Reply:
Increased the minimum slider to 16 from 1!